### ISSUE TITLE
adding/multiplying/scaling profiles for equilibrium pressure

### ISSUE THREAD

**Issue by landreman** [PlasmaControl/DESC#347]
For the j_bootstrap work I tried to specify an equilibrium pressure profile by multiplying and summing density and temperature profiles, using `ScaledProfile`, `SumProfile` and `ProductProfile`. The equilibrium solve then gave an error. Minimal example:
~~~~py
import numpy as np
from scipy.constants import elementary_charge

from desc.profiles import PowerSeriesProfile
from desc.equilibrium import Equilibrium
from desc.objectives import *

ne = PowerSeriesProfile(3.0e19 * np.array([1, -1]), modes=[0, 10])
Te = PowerSeriesProfile(2.0e3 * np.array([1, -1]), modes=[0, 2])
Ti = Te
pressure = elementary_charge * (ne * Te + ne * Ti)
print("pressure params:", pressure.params)

LM_resolution = 6
eq = Equilibrium(
    pressure=pressure,
    iota=PowerSeriesProfile([1.61]),
    Psi=np.pi,  # so B ~ 1 T                                                                                
    NFP=1,
    L=LM_resolution,
    M=LM_resolution,
    N=0,
    L_grid=2 * LM_resolution,
    M_grid=2 * LM_resolution,
    N_grid=0,
    sym=True,
)

eq.solve(
    constraints=get_fixed_boundary_constraints(),
    objective=ObjectiveFunction(objectives=ForceBalance()),
)
eq.save(__file__ + ".h5")
~~~~
Should this work, in principle? Output, using master from today:
~~~~
pressure params: (1.602176634e-19, ((array([ 3.e+19,  0.e+00,  0.e+00,  0.e+00,  0.e+00, -3.e+19]), array([ 2000., -2000.])), (array([ 3.e+19,  0.e+00,  0.e+00,  0.e+00,  0.e+00, -3.e+19]), array([ 2000., -2000.]))))
Building objective: force
Precomputing transforms
Traceback (most recent call last):
  File "/Users/mattland/Library/CloudStorage/Box-Box/work22/./20221212-09_desc_pressure_minimal_example", line 34, in <module>
    eq.solve(
  File "/Users/mattland/DESC/DESC/desc/equilibrium/equilibrium.py", line 473, in solve
    result = optimizer.optimize(
  File "/Users/mattland/DESC/DESC/desc/optimize/optimizer.py", line 290, in optimize
    objective.build(eq, verbose=verbose)
  File "/Users/mattland/DESC/DESC/desc/objectives/objective_funs.py", line 173, in build
    objective.build(eq, use_jit=self.use_jit, verbose=verbose)
  File "/Users/mattland/DESC/DESC/desc/objectives/_equilibrium.py", line 156, in build
    super().build(eq=eq, use_jit=use_jit, verbose=verbose)
  File "/Users/mattland/DESC/DESC/desc/objectives/objective_funs.py", line 598, in build
    self._set_dimensions(eq)
  File "/Users/mattland/DESC/DESC/desc/objectives/objective_funs.py", line 502, in _set_dimensions
    self._dimensions["p_l"] = eq.pressure.params.size
AttributeError: 'tuple' object has no attribute 'size'
~~~~
We can kind of get around this by adding `pressure = pressure.to_powerseries(order=12)`, but the fitting changes the coefficients a bit.

**Comment by f0uriest** [comment]
Yeah I realized this as well. The issue is that the combined profile returns a tuple of the parameters of its constituent parts rather than a single array. I'll fix that today and also add the kinetic profiles as equilibrium attributes

**Comment by f0uriest** [comment]
This should be fixed in #349. Not quite finished yet but feel free to use it for the bootstrap stuff.

**Comment by landreman** [comment]
Great, thanks!
